### DISCUSSION TITLE
Login via tsh during ci cd

### DISCUSSION THREAD

**Discussion by mustyatse** [gravitational/teleport#12864] [Q&A] [answered]
Hi, i have a problem. I need to perform login via tsh during ci cd in order to use tctl. Tsh login doesnt have possibility to pass password inline(something like `tsh login -p password`), only via prompt. Is there a way to login like this? teleport cluster is installed in k8s cluster. Sry for bad english

**Comment by webvictim** [comment, marked as answer]
You can use `tctl auth sign` as a cluster administrator to generate a longer-lived identity file that you can use with `tctl -i`. See these instructions: https://goteleport.com/docs/setup/reference/cli/#tctl-auth-sign
